### DISCUSSION TITLE
Find file related commands hangs when deleting the "~" character

### DISCUSSION THREAD

**Discussion by abougouffa** [minad/vertico#515] [Q&A]
Hello,

First, let me thank you for this awesome package which changed the way I use Emacs!

I noticed recently this issue (I'm using a fresh Emacs 30.0.60 build).

```shell
emacs -q
```

Then; `M-x package-install vertico`, `M-x vertico-mode`

Then, let's open a file under the home directory:

`~/path/to/file.txt`

Then, `C-x C-w` to save the file to another path:

In the completion, press `C-a` to go to the first character of the path `~`, move one char `C-f` then `backspace`.

```
Cursor
 |
 v
~|/path/to/file.txt

```

When deleting the `~` char, Emacs will freeze for a while (in my case, it freezes and don't release unless I press `C-g` multiple times).

See the video, recorded on an `emacs -q` instance, when I press `backspace` when `vertico-mode` is enabled, Emacs hangs for a while before actually deleting the character:

https://github.com/user-attachments/assets/5043b641-8ed9-4e67-873e-051eac3dc800

**Comment by minad** [comment]
Hi @abougouffa! Thanks for the report. Unfortunately I cannot reproduce the problem on my system. I suspect that you might have some file system/mounting issue, where accessing `/` is somehow costly. After deleting of `~`, the file completion table will try to access the root file system at `/`. If this is not the problem, I'd appreciate if you try to collect more information. You could for example set `debug-on-quit` and obtain a stack trace, such that we can determine where the code is actually hanging.
